lib/record_store/provider/dynect.rb

```diff
@@ -105,6 +106,11 @@ def build_from_api(api_record)
 
         Record.const_get(type).new(record)
       end
+
+      def synchronize(&block)
+        @mutex ||= Mutex.new
```

Comment:
Can we not define this as an instance of the class instance instead of doing the lazy loading that has the theoretical race condition?

Reply:
there is no instance :-( it's all class methods :-(

Reply:
I could assign it at load time though :-/

Reply:
eg. 
```
      @mutex = Mutex.new

      def synchronize(&block)
        @mutex.synchronize(&block)
      end
```

Reply:
👍 That's what I mean.